novawallet/Common/Network/Etherscan/ERC20/EtherscanERC20OperationFactory.swift

```diff
@@ -30,44 +43,91 @@ final class EtherscanERC20OperationFactory: EtherscanBaseOperationFactory {
         return components.url
     }
 
+    func createInfoOperation(
+        accountId: AccountId,
+        chainFormat: ChainFormat,
+        pagination: EtherscanPagination
+    ) -> BaseOperation<EtherscanERC20HistoryInfo> {
+        ClosureOperation { [weak self] in
+            guard let self else {
+                throw BaseOperationError.parentOperationCancelled
+            }
+
+            let address = try accountId.toAddress(using: chainFormat)
+            let ethereumAddress = address.toEthereumAddressWithChecksum() ?? address
+
+            let info = EtherscanERC20HistoryInfo(
+                address: ethereumAddress,
+                contractaddress: contractAddress,
+                page: pagination.page,
+                offset: pagination.offset
+            )
+
+            return info
+        }
+    }
+
     func createFetchWrapper(
-        for info: EtherscanERC20HistoryInfo
+        for accountId: AccountId,
+        chainFormat: ChainFormat,
+        pagination: EtherscanPagination
     ) -> CompoundOperationWrapper<WalletRemoteHistoryData> {
-        guard let url = buildUrl(for: info) else {
-            return CompoundOperationWrapper.createWithError(NetworkBaseError.invalidUrl)
+        let infoOperation = createInfoOperation(
```

Comment:
Don't see any purpose to wrap that into an operation and add dependency on operation manager